### PR TITLE
Fix issue when trying to modify user profiles

### PR DESCRIPTION
**Pull request by Techassi**
This PR introduces:

- Optimized user profile selection and removing duplicate code
- Fix issue while modifying user profiles

## Fix

When trying to modify a user the changes were rejected because a user with the same alias already exists. This fixes the issue by looking at the index of the changed user. If we look at the same user we skip validation.

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -197,12 +197,11 @@ func (c *Config) SelectUser(index int) (*models.User, error) {
 
 // SelectDefaultUser returns the default user or nil if there is no default user
 func (c *Config) SelectDefaultUser() (*models.User, error) {
-	for _, user := range c.Users {
-		if user.Alias == constants.DefaultAlias {
-			return &user, nil
-		}
+	user, err := c.SelectUserByAlias("default")
```
Comment (matsuyoshi30) [inline internal/config/config.go]: How about using `internal/constants.DefaultAlias` or deleting that file? DefaultAlias constant would be used here only.

**Comment by matsuyoshi30** [comment]
Hi @Techassi. Sorry for the delay and thanks for your contribution again! Check my comment, please.

File: internal/config/config.go (hunk 1 above)
Comment (Techassi) [reply internal/config/config.go]: Good catch. I guess the "correct" way to do this is using the constant value. If we use the constant in more places at some point in the future this is probably better maintainable. Let me know your thoughts about this.

Comment (Techassi) [reply internal/config/config.go]: Update on this? @matsuyoshi30

Comment (matsuyoshi30) [reply internal/config/config.go]: I agree it would be better to use the constant value 👍 .

Comment (Techassi) [reply internal/config/config.go]: Okay great! Will update this.

Comment (Techassi) [reply internal/config/config.go]: Resolved with 7fa366b

**Comment by matsuyoshi30** [comment]
Looks great! Thanks for your contribution again @Techassi!

**Pull request merged**